src/miner_keys.erl

```diff
@@ -128,27 +198,68 @@ key_config() ->
         KC -> KC
     end.
 
+-spec libp2p_to_gateway_key(libp2p_crypto:key_map()) -> libp2p_crypto:key_map().
+libp2p_to_gateway_key(#{secret := {ecc_compact, PrivKey}}) ->
+    {ok, #{
+        secret => {ecc_compact, PrivKey#'ECPrivateKey'{publicKey = <<>>}},
+        public => {ecc_compact, <<>>}
+    }}.
+
 %% @doc prints the public hotspot and onboadring key in a file:consult
 %% friendly way to stdout. This is used by other services (like
 %% gateway_config) to get read access to the public keys
 print_keys(_) ->
     #{
-       pubkey := PubKey,
-       onboarding_key := OnboardingKey
-     } = keys(),
-    MaybeB58 = fun(undefined) -> undefined;
-                  (Key) -> libp2p_crypto:pubkey_to_b58(Key)
-               end,
-    Props = [{pubkey, MaybeB58(PubKey)},
-             {onboarding_key, OnboardingKey}
-            ] ++ [ {animal_name, element(2, erl_angry_purple_tiger:animal_name(libp2p_crypto:pubkey_to_b58(PubKey)))} || PubKey /= undefined ],
+        pubkey := PubKey,
+        onboarding_key := OnboardingKey
+    } = keys(),
+    MaybeB58 = fun
+        (undefined) -> undefined;
+        (Key) -> libp2p_crypto:pubkey_to_b58(Key)
+    end,
+    Props =
+        [
+            {pubkey, MaybeB58(PubKey)},
+            {onboarding_key, OnboardingKey}
+        ] ++
+            [
+                {animal_name,
+                    element(
+                        2, erl_angry_purple_tiger:animal_name(libp2p_crypto:pubkey_to_b58(PubKey))
+                    )}
+             || PubKey /= undefined
+            ],
     lists:foreach(fun(Term) -> io:format("~tp.~n", [Term]) end, Props),
     rpc_ok.
 
+key_proplist_to_uri(Props) ->
+    Bus = proplists:get_value(bus, Props, "i2c-1"),
+    Address = proplists:get_value(address, Props, 16#60),
+    KeySlot = proplists:get_value(key_slot, Props, 0),
+    Network = application:get_env(blockchain, network, mainnet),
+    PortAndQuery = lists:flatten(io_lib:format(":~p?slot=~p&network=~s", [Address, KeySlot, Network])),
+    "ecc://" ++ Bus ++ PortAndQuery.
+
+key_uri_to_proplist(Uri) ->
+    #{host := Bus, port := Address, query := Query} = uri_string:parse(Uri),
+    QueryList = lists:foldl(fun({"slot", Slot}, Acc) ->
+                                   [{key_slot, list_to_integer(Slot)} | Acc];
+                               ({"network", Network}, Acc) ->
+                                   [{network, Network} | Acc];
+                               (_, Acc) -> Acc
+                            end, [], uri_string:dissect_query(Query)),
+    [{bus, Bus}, {address, Address}] ++ QueryList.
 %%
 %% Utilities
 %%
 
+key_proplist_to_map(Options) when is_list(Options) ->
```

Comment:
You may want to match that this is a list of tuples, not a string?

Reply:
like `key_proplist_to_map([{_, _} | _] = Options) ->` or is there a different guard i'm missing?